src/cljs/main/broadfcui/page/workspace/method_configs/launch_analysis.cljs

```diff
@@ -69,16 +83,7 @@
                                                     :workflow-count (common/count-workflows
                                                                       e (:root-entity-type props)))})))}]]
    (style/create-form-label "Define Expression")
-   (let [disabled (= (:root-entity-type props) (get-in @state [:selected-entity :type]))]
-     (style/create-text-field {:placeholder "leave blank for default"
-                               :style {:width "100%"
-                                       :backgroundColor (when disabled (:background-light style/colors))}
-                               :disabled disabled
-                               :value (if disabled
-                                        "Disabled - selected entity is of root entity type"
-                                        (:expression @state))
-                               :onChange #(let [text (-> % .-target .-value clojure.string/trim)]
-                                            (swap! state assoc :expression text))}))
+   [DefineExpressionField props state]
```

Comment:
Moved to its own component because jquery3+react complained about updates from uncontrolled to controlled components: https://facebook.github.io/react/docs/forms.html#controlled-components